dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java

```diff
@@ -840,29 +841,45 @@ public IHTMLPage findByIdLanguageFallback(final Identifier identifier, final lon
     public IHTMLPage findByIdLanguageFallback(final String identifier, final long tryLang, final boolean live, final User user, final boolean respectFrontEndPermissions)
             throws DotDataException, DotSecurityException {
 
-
         IHTMLPage htmlPage;
-
         Contentlet contentlet;
 
         try {
             contentlet = APILocator.getContentletAPI().findContentletByIdentifier(identifier, live, tryLang,
                     user, respectFrontEndPermissions);
             htmlPage = APILocator.getHTMLPageAssetAPI().fromContentlet(contentlet);
 
-        } catch (Exception dse) {
-            if (APILocator.getLanguageAPI().canDefaultPageToDefaultLanguage() && tryLang != APILocator.getLanguageAPI().getDefaultLanguage().getId()) {
+        } catch (DotContentletStateException dse) {
+            htmlPage = findPageInDefaultLanguage(identifier, tryLang, live, user, respectFrontEndPermissions, dse);
+
+        }
+
+        return htmlPage;
+    }
+
+    private IHTMLPage findPageInDefaultLanguage(String identifier, long tryLang, boolean live, User user,
```

Comment:
If I use the new method `findPageInDefaultLanguage` passing as parameter the default language id it will return null right?, maybe we could change the name of the method as is not really just searching by the default language and could be confusing if someone else tries to use it.